.github/workflows/check_java.yml

```diff
@@ -0,0 +1,45 @@
+# Runs tests and formatting requirements
+name: Java Checks
+
+on:
+  push:
+    branches:
+      - master
+    paths: ['src/main/java/**', 'src/test/**', 'src/api/java/**', 'gradle/**', '**.gradle', 'gradle.properties',
+            'gradlew**', 'src/main/resources/*_at.cfg']
+  pull_request:
+    paths: ['src/main/java/**', 'src/test/**', 'src/api/java/**', 'gradle/**', '**.gradle', 'gradle.properties',
+            'gradlew**', 'src/main/resources/*_at.cfg']
+
+concurrency:
+  group: checks-${{ github.head_ref || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: Tests
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Build
+        uses: ./.github/actions/build_setup
+
+      - name: Run Tests with Gradle
+        run: ./gradlew test --warning-mode all --build-cache
+
+  formatting:
+    name: Formatting
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Setup Build
+        uses: ./.github/actions/build_setup
+
+      - name: Run Spotless Formatting with Gradle
+        run: ./gradlew spotlessCheck --warning-mode all --build-cache
```

Comment:
Can we exclude some of the provided paths from Spotless formatting? We don't really need to Spotless format the gradle files, and actually should not spotless format build.gradle. Also, we should not run spotless on `/src/api/java`, as it will fail due to using annotations we prohibited, Nonnull, etc, but we should not change those, as we want copies of the class to match with how it is in the mod source code.